lib/pbench/server/api/resources/query_apis/index_search.py

```diff
@@ -0,0 +1,159 @@
+                    ["@timestamp", "run.controller", "run.name", ...]
+                    The fields controls what fields are returned by the query, and it has no effect
+                    on which fields are searched.
+        """
+        user = json_data.get("user")
+        start = json_data.get("start")
+        end = json_data.get("end")
+
+        # If the search_term parameter is not specified we will return all the Elasticsearch hits
+        search_term = json_data.get("search_term", "")
+        # If no fields are specified, the query will return all the fields from Elasticsearch hits
+        selected_fields = json_data.get("fields", [])
+
+        # We need to pass string dates as part of the Elasticsearch query; we
+        # use the unconverted strings passed by the caller rather than the
+        # adjusted and normalized datetime objects for this.
+        start_arg = f"{start:%Y-%m-%d}"
+        end_arg = f"{end:%Y-%m-%d}"
+
+        self.logger.info(
+            "Search query for user {}, prefix {}: ({} - {}) on query: {}",
+            user,
+            self.prefix,
+            start,
+            end,
+            search_term,
+        )
+
+        uri_fragment = self._gen_month_range("run", start, end)
+        self.logger.info("fragment, {}", uri_fragment)
+        return {
+            "path": f"/{uri_fragment}/_search",
+            "kwargs": {
+                "json": {
+                    "query": {
+                        "bool": {
+                            "filter": [
+                                {"term": self._get_user_term(json_data)},
+                                {
+                                    "range": {
+                                        "@timestamp": {"gte": start_arg, "lte": end_arg}
+                                    }
+                                },
+                                {"query_string": {"query": f"*{search_term}*"}},
+                            ]
+                        }
+                    },
+                    "sort": [{"@timestamp": {"order": "desc"}}],
+                    "_source": {"include": selected_fields},
+                },
+                "params": {"ignore_unavailable": "true"},
+            },
+        }
+
+    def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
+        """
+        Returns a summary of the returned Elasticsearch query results, showing
+        the list of dictionaries with user selected fields from request json as keys
+
+        [
+            {
+                "id": "1c25e9f5b5dfc1ffb732931bf3899878",
+                "@timestamp": "2021-07-12T22:44:19.562354",
+                "run": {
+                    "controller": "dhcp31-171.perf.lab.eng.bos.redhat.com",
+                    "name": "pbench-user-benchmark_npalaska-dhcp31-171_2021.07.12T22.44.19",
+                },
+                "@metadata": {
+                    "controller_dir": "dhcp31-171.perf.lab.eng.bos.redhat.com"
+                }
```

Comment:
It might be useful here to clarify which of these fields are "forced" (e.g., `id` which we add) vs the others which appear only based on the client-selected fields.